frontend/public/components/catalog-items.jsx

```diff
@@ -1,48 +1,204 @@
+import {categorizeItems} from './utils/categorize-catalog-items';
 
-const CatalogItem = ({item}) => {
-  const catalogTile = <CatalogTile
-    id={item.obj.metadata.uid}
-    key={item.obj.metadata.uid}
-    title={item.tileName}
-    iconImg={item.tileImgUrl}
-    iconClass={item.tileIconClass ? `icon ${normalizeIconClass(item.tileIconClass)}` : null}
-    vendor={item.tileProvider ? `Provided by ${item.tileProvider}` : null}
-    description={item.tileDescription}
-  />;
-  return item.href ? <Link className="co-catalog-item-tile" to={item.href}>{catalogTile}</Link> : catalogTile;
-};
+export class CatalogTileViewPage extends React.Component {
+  constructor(props) {
+    super(props);
 
-CatalogItem.displayName = 'CatalogItem';
-
-CatalogItem.propTypes = {
-  item: PropTypes.shape({
-    obj: PropTypes.object.isRequired,
-    kind: PropTypes.string.isRequired,
-    tileName: PropTypes.string.isRequired,
-    tileIconClass: PropTypes.string,
-    tileImgUrl: PropTypes.string,
-    tileDescription: PropTypes.string.isRequired,
-    tileProvider: PropTypes.string,
-  }).isRequired
-};
+    this.state = {
+      showAllItemsForCategory: null,
+      activeTabs: ['all'], // array of tabs [main category, sub-category]
+    };
+
+    if (props.items) {
+      this.allCategories = categorizeItems(props.items);
+      _.assign(this.state, {
+        currentCategories: this.allCategories,
+        numItems: _.first(this.allCategories).numItems,
+      });
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const {items} = this.props;
+
+    if (items !== nextProps.items) {
+      this.allCategories = categorizeItems(items);
+      this.setState({
+        currentCategories: this.allCategories,
+        numItems: _.first(this.allCategories).numItems
+      });
+    }
+  }
+
+  isAllTabActive() {
+    const { activeTabs } = this.state;
+    return activeTabs[0] === 'all';
+  }
+
+  activeTabIsSubCategory(subCategories) {
+    const { activeTabs } = this.state;
+    return activeTabs.length !== 2 ? false : _.find(subCategories, ['id', activeTabs[1]]) !== undefined;
+  }
+
+  isActiveTab(categoryId) {
+    const { activeTabs } = this.state;
+    return activeTabs.length === 2 ? categoryId === activeTabs[1] : categoryId === activeTabs[0];
+  }
+
+  doesHaveActiveDescendants(categoryId) {
+    const { activeTabs } = this.state;
+    return activeTabs[0] === categoryId;
+  }
+
+  renderTabs(category, parentCategoryId=null){
+    if (!category.numItems) {
+      return;
+    }
+    return <VerticalTabs.Tab
+      key={category.id}
+      title={category.label}
+      active={this.isActiveTab(category.id)}
+      onActivate={() => this.onActivateTab(parentCategoryId ? [parentCategoryId, category.id] : [category.id])}
+      hasActiveDescendant={this.doesHaveActiveDescendants(category.id)}
+      shown={category.id === 'all'}>
+      {category.subCategories && category.numItems && <VerticalTabs key={`${category.id}-tabs`}
```

Comment:
category.numItems is always true because we check it above

I don't think you need the `-tabs` on then end of key